src/utils/middleware.ts

```diff
@@ -0,0 +1,24 @@
+import { ResponseError } from '../pinecone-generated-ts-fetch';
+import { handleApiError } from '../errors';
+
+export const middleware = [
+  {
+    onError: async (context) => {
```

Comment:
**praise:** Huge praise for all this cleanup, I feel like errors are much more straightforward to reason around now, and it's really helpful to pull all the testing into a more centralized place rather than having it spread across all the individual methods.